vpr/src/draw/search_bar.cpp

```diff
@@ -336,34 +406,111 @@ void warning_dialog_box(const char* message) {
+    } else if (searchType == "Net Name") {
+        gtk_entry_completion_set_model(completion, netNames);
+        gtk_entry_set_completion(searchBar, completion);
+        gtk_widget_show(keyLengthSpinButton);
+        gtk_widget_show(keyLengthLabel);
+    } else { //setting to null if option does not require auto-complete
+        gtk_widget_hide(keyLengthSpinButton);
+        gtk_widget_hide(keyLengthLabel);
         gtk_entry_set_completion(searchBar, nullptr);
     }
 }
 
 /**
  * @brief loads block names into gtk list store item used for completion
  * 
- * @param app ezgl application
+ * @param app ezgl application used for ui
  */
 void load_block_names(ezgl::application* app) {
     auto blockStorage = GTK_LIST_STORE(app->get_object("BlockNames"));
     auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& atom_ctx = g_vpr_ctx.atom();
     GtkTreeIter iter;
-    //Getting and storing all block names
     for (ClusterBlockId id : cluster_ctx.clb_nlist.blocks()) {
         gtk_list_store_append(blockStorage, &iter);
         gtk_list_store_set(blockStorage, &iter,
                            0, (cluster_ctx.clb_nlist.block_name(id)).c_str(), -1);
     }
+    for (AtomBlockId id : atom_ctx.nlist.blocks()) {
+        gtk_list_store_append(blockStorage, &iter);
+        gtk_list_store_set(blockStorage, &iter,
+                           0, (atom_ctx.nlist.block_name(id)).c_str(), -1);
+    }
+}
+
+/**
+ * @brief loads net names into gtk list store item used for completion
+ * 
+ * @param app ezgl application used for ui
+ */
+void load_net_names(ezgl::application* app) {
+    auto netStorage = GTK_LIST_STORE(app->get_object("NetNames"));
+    auto& atom_ctx = g_vpr_ctx.atom();
+    GtkTreeIter iter;
+    //Loading net names
+    for (AtomNetId id : atom_ctx.nlist.nets()) {
+        gtk_list_store_append(netStorage, &iter);
+        gtk_list_store_set(netStorage, &iter,
+                           0, (atom_ctx.nlist.net_name(id)).c_str(), -1);
+    }
+}
+
+/**
+ * @brief A new matching function used for the wildcard search options
+ * Will be slower
+ * 
+ * @param completer the GtkEntryCompletion being used
+ * @param key a normalized and case-folded key representing the text
+ * @param iter GtkTreeIter pointing at the current entry being compared
+ * @param user_data null
+ * @return gboolean 
+ */
+gboolean customMatchingFunction(
+    GtkEntryCompletion* completer,
+    const gchar* key,
+    GtkTreeIter* iter,
+    gpointer /*user data*/
+) {
+    GtkTreeModel* model = gtk_entry_completion_get_model(completer);
+    const gchar* text;
+    gtk_tree_model_get(model, iter, 0, &text, -1);
+    //Removing case information
+    g_utf8_casefold(text, -1);
+    g_utf8_normalize(text, -1, G_NORMALIZE_DEFAULT);
+    std::string cppText(text);
+    //If substring not found, returning false;
+    return (cppText.find(key, 0) != std::string::npos);
+}
+
+/**
+ * @brief Updates min. key length when the value in spin button changes
+ */
```

Comment:
I'd give a bit more context on this one -- explain what the key length is for. Can also say this is a callback for when the user clicks on an arrow to update the spin button controlling the length (which I think is the case).